pkg/subctl/cmd/gather.go

```diff
@@ -0,0 +1,120 @@
+/*
+© 2021 Red Hat, Inc. and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	gatherType   string
+	gatherModule string
+)
+
+var gatherModuleFlags = map[string]bool{
+	"connectivity": false,
+	"discovery":    false,
+	"broker":       false,
+	"operator":     false,
+}
+
+var gatherTypeFlags = map[string]bool{
+	"logs":      false,
+	"resources": false,
+}
+
+func init() {
+	addKubeconfigFlag(gatherCmd)
+	addGatherFlags(gatherCmd)
+	rootCmd.AddCommand(gatherCmd)
+}
+
+func addGatherFlags(gatherCmd *cobra.Command) {
+	gatherCmd.Flags().StringVar(&gatherType, "type", strings.Join(getAllTypeKeys(), ","), "comma-separated list of data types to gather")
+	gatherCmd.Flags().StringVar(&gatherModule, "module", strings.Join(getAllModuleKeys(), ","), "comma-separated list of components for which to gather data")
+}
+
+var gatherCmd = &cobra.Command{
+	Use:   "gather <kubeConfig1>",
```

Comment:
Sorry, I didn't notice this before. 

In other modules, except for verify (which needs multiple kubeconfigs),
we use just the --kubeconfig.

I think that in some cases customers don't even know where the kubeconfig could be (for example after oc login)

Should we change to --kubeconfig instead? 
may be even accept --namespace for when submariner is not in the default namespace?

Reply:
> Sorry, I didn't notice this before.
> 
> In other modules, except for verify (which needs multiple kubeconfigs),
> we use just the --kubeconfig.
> 
> I think that in some cases customers don't even know where the kubeconfig could be (for example after oc login)
> 
> Should we change to --kubeconfig instead?
> may be even accept --namespace for when submariner is not in the default namespace?

Later we will want to add support to pull data from multiple clusters. So thought let us use `kubeConfig1` so that later on we can add more to it. As for namespace, don't we store it in Submariner CR? What if we read it off there and don't leave user any room for error?

Reply:
Changed to `--kubeconfig` for now. Will revisit when we support multiple clusters.